sormas-ui/src/main/java/de/symeda/sormas/ui/externalsurveillanceservice/ExternalSurveillanceShareComponent.java

```diff
@@ -77,16 +84,27 @@ private HorizontalLayout createHeader(String entityName, Runnable sendHandler, R
 			headerLayout.addComponent(sendButton);
 			headerLayout.setExpandRatio(sendButton, 1);
 			headerLayout.setComponentAlignment(sendButton, Alignment.MIDDLE_RIGHT);
-
+		}
+		if (isVisibleDeleteButton(caseUuid)) {
 			Button deleteButton = ButtonHelper.createIconButton("", VaadinIcons.TRASH, e -> deleteHandler.run(), ValoTheme.BUTTON_ICON_ONLY);
 			headerLayout.addComponent(deleteButton);
 			headerLayout.setComponentAlignment(deleteButton, Alignment.MIDDLE_RIGHT);
 			headerLayout.setWidth(100, Unit.PERCENTAGE);
 		}
-		
+
 		return headerLayout;
 	}
 
+	private boolean isVisibleDeleteButton(String caseUuid) {
+		CaseDataDto caseDataDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(caseUuid);
+		boolean isSharedCase = FacadeProvider.getExternalShareInfoFacade().isSharedCase(caseUuid);
+
+		if (caseDataDto.isDontShareWithReportingTool() && !isSharedCase) {
```

Comment:
Won't this hide the button for cases that HAVE been shared, but are not annotated with "don't share with reporting tool"? Wouldn't it be better to use || instead of && here?

(I'm not that familiar with this functionality, so this is more of a question than a definitive change request).

Reply:
no in that scenario the button will not be hidden beccause the the dontShareWithReportingTool will be false . I  implemented to work like this
- if the case has dontShareWithReportingTool checkbox on true and the case was shared (this is the scenario for old data) then the button will be visible
- if the case has dontShareWithReportingTool checkbox on true and the case was not shared  then the button will be hidden , as well as the button to send to reporting tool . In this case the use is no longer allowed to communicate with reporting tool
- if the case has dontShareWithReportingTool checkbox on false and the case was shared , the delete button will be visible . In this case the send to reporting tool button is visible as well
- if the case has dontShareWithReportingTool checkbox on false and the case was not shared , the delete button will be visible . In this case the send to reporting tool button is visible as well

dontShareWithReportingTool        shared                deleteButtonVisible               sendButtonVisible
true                               true                    true                              false
true                               false                   false                             false
false                              true                    true                              true 
false                              false                   true                              true